extensions/sql-assessment/src/engine.ts

```diff
@@ -56,6 +56,7 @@ export class AssessmentEngine {
 	public async initialize(connectionId: string) {
 		this.connectionUri = await azdata.connection.getUriForConnection(connectionId);
 		this.connectionProfile = await azdata.connection.getCurrentConnection();
+		this.historicalRecords = undefined;
```

Comment:
Why not just initialize to an empty array instead?

Reply:
to distinguish the case when there are no historical results, so we don't need to search for history file every time.